cpp/src/io/parquet/writer_impl.hpp

```diff
@@ -93,6 +93,14 @@ class writer::impl {
    */
   void init_state();
 
+  /**
+   * @brief Updates writer-level statistics with data from the current table.
+   *
+   * @param compression_stats Optional compression statistics from the current table
+   */
+  void update_compression_statistics(
```

Comment:
Similar question, can we make this a free function?

Reply:
It changes the internal state of the `impl` object, I think it fits better as a member function.